### PR TITLE
Convert repeated tests to JUnit 5 @RepeatableTest

### PR DESCRIPTION
**Pull request by garyrussell**
- add lifecycle to `@LogLevels` to avoid adjusting the log on each iteration
- with `@RepeatableTest` there is a template context between the class and method contexts

### CODE REVIEW

File: spring-rabbit-junit/src/main/java/org/springframework/amqp/rabbit/junit/LogLevels.java [hunk 1]
```diff
@@ -57,4 +58,11 @@
 	 */
 	String level() default "DEBUG";
```
Comment (artembilan) [inline spring-rabbit-junit/src/main/java/org/springframework/amqp/rabbit/junit/LogLevels.java]: You know, I've checked a `Level.toLevel()` and it defaults to `DEBUG`. For now... It might be changed eventually.
So, let's revert this change to the empty string and really rely on the default in the target Log4J impl!

File: spring-rabbit-junit/src/main/java/org/springframework/amqp/rabbit/junit/LogLevels.java [hunk 2]
```diff
@@ -57,4 +58,11 @@
...
 
+	/**
+	 * When used at the class level, set this to PER_CLASS to set/restore
+	 * the levels once only. Useful with {@code @RepeatedTest}.
+	 * @return the Lifecycle.
+	 */
+	TestInstance.Lifecycle lifecycle() default TestInstance.Lifecycle.PER_METHOD;
```
Comment (artembilan) [inline spring-rabbit-junit/src/main/java/org/springframework/amqp/rabbit/junit/LogLevels.java]: Can't this be per class by default? In most cases we really use it on the class level without considering it to be changed in between. Well, essentially it does just an idempotent work between methods anyway, so, I even would consider do not have this option at all. On class level it is for the whole class; on method level - just for method and has a precedence over the class one.

What am I missing with this option, please?

Thanks

File: spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitTemplatePerformanceIntegrationTests.java [hunk 3]
```diff
@@ -44,87 +49,78 @@
  * @since 1.0
  *
  */
+@RabbitAvailable(queues = RabbitTemplatePerformanceIntegrationTests.ROUTE)
+@LongRunning
```
Comment (artembilan) [inline spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitTemplatePerformanceIntegrationTests.java]: Is it still really *long-runing* ?

File: spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/MessageListenerRecoveryRepeatIntegrationTests.java [hunk 4]
```diff
@@ -57,13 +56,19 @@
  * @author Gary Russell
  *
  */
+@RabbitAvailable(queues = MessageListenerRecoveryRepeatIntegrationTests.TEST_QUEUE, purgeAfterEach = false)
+@LogLevels(level = "ERROR", lifecycle = Lifecycle.PER_CLASS, classes = { RabbitTemplate.class,
+		ConditionalRejectingErrorHandler.class,
+		SimpleMessageListenerContainer.class, BlockingQueueConsumer.class,
+		MessageListenerRecoveryRepeatIntegrationTests.class })
+@TestInstance(Lifecycle.PER_CLASS)
```
Comment (artembilan) [inline spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/MessageListenerRecoveryRepeatIntegrationTests.java]: Shouldn't we consider to have this option enabled globally for all the test classes?
I somehow believe that Spring Test Framework makes test instances really per class anyway...

File: spring-rabbit-junit/src/main/java/org/springframework/amqp/rabbit/junit/LogLevels.java (hunk 2 above)
Comment (garyrussell) [reply spring-rabbit-junit/src/main/java/org/springframework/amqp/rabbit/junit/LogLevels.java]: I kept it this way so we get the nice delimiter in the log between tests; but I have another idea. Stay tuned...

File: spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitTemplatePerformanceIntegrationTests.java (hunk 3 above)
Comment (garyrussell) [reply spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitTemplatePerformanceIntegrationTests.java]: 7 seconds total.

File: spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/MessageListenerRecoveryRepeatIntegrationTests.java (hunk 4 above)
Comment (garyrussell) [reply spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/MessageListenerRecoveryRepeatIntegrationTests.java]: >I somehow believe

I didn't believe so (and I just tested it) - the context is cached but you get a new instance per test method.

We can try setting it globally, but I'd rather wait until all conversion is done before trying it.

Comment (artembilan) [reply spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/MessageListenerRecoveryRepeatIntegrationTests.java]: Yeah... Got it. Thank you!
Yep! Let's deffer it to the later phase!

Looking forward for your final solution for that `lifecycle` concern!
